tests/components/vacuum/test_reproduce_state.py

```diff
@@ -0,0 +1,127 @@
+"""Test reproduce state for Vacuum."""
+from homeassistant.components.vacuum import (
+    ATTR_FAN_SPEED,
+    STATE_CLEANING,
+    STATE_DOCKED,
+    STATE_RETURNING,
+)
+from homeassistant.const import STATE_IDLE, STATE_OFF, STATE_ON, STATE_PAUSED
+from homeassistant.core import State
+
+from tests.common import async_mock_service
+
+FAN_SPEED_LOW = "low"
+FAN_SPEED_HIGH = "high"
+
+
+async def test_reproducing_states(hass, caplog):
+    """Test reproducing Vacuum states."""
+    hass.states.async_set("vacuum.entity_off", STATE_OFF, {})
+    hass.states.async_set("vacuum.entity_on", STATE_ON, {})
+    hass.states.async_set(
+        "vacuum.entity_on_fan", STATE_ON, {ATTR_FAN_SPEED: FAN_SPEED_LOW}
+    )
+    hass.states.async_set("vacuum.entity_cleaning", STATE_CLEANING, {})
+    hass.states.async_set("vacuum.entity_docked", STATE_DOCKED, {})
+    hass.states.async_set("vacuum.entity_idle", STATE_IDLE, {})
+    hass.states.async_set("vacuum.entity_returning", STATE_RETURNING, {})
+    hass.states.async_set("vacuum.entity_paused", STATE_PAUSED, {})
+
+    turn_on_calls = async_mock_service(hass, "vacuum", "async_turn_on")
```

Comment:
You're mocking the wrong service. Remove async_ from all.

Reply:
Thank you very much. Sorry, I've thought this should be the related method names